lib/std/smartptrs.nim

```diff
@@ -0,0 +1,133 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2021 Nim contributors
+#
```

Comment:
> {.error: "smartptrs requires --threads:on option.".}

(moving to here as other thread was different topiic)

that's not good enough; it'll force client code to use --threads:on just so they can use some library that happens to use `smartptrs` somewhere, which can negatively impact performance.

And when you combine this with things like https://github.com/dom96/httpbeast/commit/da518bb6f66fc6d91f7406ba0be6eae4185c24d0#diff-09eca2bfdf8a9ad44e6134304f3a4a656ad73f39790651cd1f8ed80db4d1d774R325, you end up with arbitrary restrictions, eg: can't use `reusePort=false` in your project if it depends transitively on std/smartptrs.

We can discuss what's currently blocking supporting threads:off in this module and then fix it.

Reply:
Ok I can redo https://github.com/nim-lang/fusion/pull/37, however as I followed the discussion on #17333 it's not recommended to use smartptrs for everyday Nim code, one should use ref. It's a mechanism, that's not enough by itself mind you, to share data between threads. The docs I'm writing will be very explicit about it.

Reply:
> Ok I can redo nim-lang/fusion#37, 

great.

Reply:
> There would be a serious usability / compatibility issues otherwise.

Like what exactly? have you used it in a project? and absolutely needed --threads:off? Do you use locks with without threads as well?